Makefile

```diff
@@ -28,49 +30,31 @@ endef
 export POM_XML_FOR_GETTING_DEPENDENT_JARS
 download_jars:
 	command -v mvn >/dev/null 2>&1 || { echo >&2 "Maven not installed. Install maven!"; exit 1; }
-	mkdir -p $(JAR_DIR)/ /tmp/kinesis-to-firehose
-	echo $$POM_XML_FOR_GETTING_DEPENDENT_JARS > /tmp/kinesis-to-firehose/pom.xml
-	cd /tmp/kinesis-to-firehose && mvn dependency:copy-dependencies
-	cp /tmp/kinesis-to-firehose/target/dependency/* $(JAR_DIR)/
+	mkdir -p $(JAR_DIR) $(TMP_DIR)
+	echo $$POM_XML_FOR_GETTING_DEPENDENT_JARS > $(TMP_DIR)/pom.xml
+	cd $(TMP_DIR) && mvn dependency:copy-dependencies
+	mv $(TMP_DIR)/target/dependency/* $(JAR_DIR)/
+	# Download the STS jar file for supporting IAM Roles
+	ls $(JAR_DIR)/aws-java-sdk-core-*.jar | sed -e "s/.*-sdk-core-//g" | sed -e "s/\.jar//g" > /tmp/version.txt
+	curl -o $(JAR_DIR)/aws-java-sdk-sts-`cat /tmp/version.txt`.jar http://central.maven.org/maven2/com/amazonaws/aws-java-sdk-sts/`cat /tmp/version.txt`/aws-java-sdk-sts-`cat /tmp/version.txt`.jar
+	rm -r $(TMP_DIR)
```

Comment:
Might it be worth refactoring the kcl base stuff into a Makefile ~`kcl.mk`? This could be included in the parent repo, and would refactor all the common makefile setup stuff a la `golang.mk`